main.c

```diff
@@ -74,137 +129,231 @@ int main(int argc, char **argv) {
+            can_msg_t sensor_msg;
+            build_analog_data_msg(millis(), SENSOR_PRESSURE_FUEL, pressure_fuel_psi, &sensor_msg);
+            txb_enqueue(&sensor_msg);
+        }
+#endif
+
+#if PRES_CC_TIME_DIFF_ms
+        if (millis() - last_pres_cc_millis > PRES_CC_TIME_DIFF_ms) {
+            last_pres_cc_millis = millis();
+            // uint16_t pressure_cc_psi = update_pressure_psi_low_pass(pres_cc);
+            uint16_t pressure_cc_psi = get_pressure_4_20_psi(pres_cc);
+            can_msg_t sensor_msg;
+            build_analog_data_msg(millis(), SENSOR_PRESSURE_CC, pressure_cc_psi, &sensor_msg);
+            txb_enqueue(&sensor_msg);
+        }
+#endif
+            
+#if HALLSENSE_FILL_TIME_DIFF_ms 
+            if (millis() - last_hallsense_fill_millis > HALLSENSE_FILL_TIME_DIFF_ms) {
+                last_hallsense_fill_millis = millis();
+                uint16_t hallsense_fill_flux = get_hall_sensor_reading(hallsense_fill);
+                can_msg_t sensor_msg;
+                build_analog_data_msg(millis(), SENSOR_HALL_FILL, hallsense_fill_flux, &sensor_msg);
+                txb_enqueue(&sensor_msg);
+            }
+#endif
+
+#if HALLSENSE_FUEL_TIME_DIFF_ms
+        if (millis() - last_hallsense_fuel_millis > HALLSENSE_FUEL_TIME_DIFF_ms) {
+            last_hallsense_fuel_millis = millis();
+            uint16_t hallsense_fuel_flux = get_hall_sensor_reading(hallsense_fuel);
+            can_msg_t sensor_msg;
+            build_analog_data_msg(millis(), SENSOR_HALL_FUEL_INJ, hallsense_fuel_flux, &sensor_msg);
+            txb_enqueue(&sensor_msg);
+        }
+#endif
+
+#if HALLSENSE_OX_TIME_DIFF_ms
+        if (millis() - last_hallsense_ox_millis > HALLSENSE_OX_TIME_DIFF_ms) {
+            last_hallsense_ox_millis = millis();
+            uint16_t hallsense_ox_flux = get_hall_sensor_reading(hallsense_ox);
+            can_msg_t sensor_msg;
+            build_analog_data_msg(millis(), SENSOR_HALL_OX_INJ, hallsense_ox_flux, &sensor_msg);
+            txb_enqueue(&sensor_msg);
+        }
+#endif
+
+#if VENT_TEMP_TIME_DIFF_ms
+        if (millis() - last_vent_temp_millis > VENT_TEMP_TIME_DIFF_ms) {
+            last_vent_temp_millis = millis();
+
+            uint16_t temperature_c = get_temperature_c(temp_vent);
 
             can_msg_t sensor_msg;
             build_analog_data_msg(millis(), SENSOR_VENT_TEMP, temperature_c, &sensor_msg);
             txb_enqueue(&sensor_msg);
         }
 #endif
 
+#if PRES_OX_TIME_DIFF_ms
+        if (millis() - last_pres_ox_millis > PRES_OX_TIME_DIFF_ms) {
+            last_pres_ox_millis = millis();
+            uint16_t pressure_ox_psi = update_pressure_psi_low_pass(pres_ox);
+            can_msg_t sensor_msg;
+            build_analog_data_msg(millis(), SENSOR_PRESSURE_OX, pressure_ox_psi, &sensor_msg);
+            txb_enqueue(&sensor_msg);
+        }
+#endif
         // send any queued CAN messages
         txb_heartbeat();
     }
```

Comment:
Github wont let me comment lower but remove the random send_status_ok thats after the while loop, we'll never get there.